### PR TITLE
SNOW-1748333 Fix Iceberg decimal type schema parser

### PR DESCRIPTION
**Pull request by sfc-gh-alhuang**
## Summary
This PR addresses the issue where the scanner encounters an error when scanning delta-encoded decimal values with a precision greater than 18.

## Changes
1. Removed `sfVer` metadata in Iceberg mode: This prevents the scanner from applying optimizations specific to Snowflake-written Parquet files, which was the root cause of the issue with scanning.
2. Add tests for multiple random generated decimals.

### CODE REVIEW

File: src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java
Comment (sfc-gh-hmadan) [inline src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java:90]: 1. from line 101 it looks like we are adding the logical/physical type info to metadata for iceberg mode too (I thought from our slack conv that we weren't adding it in) ?
2. Can we scope this change to only kick in when serialization policy = optimized?

File: src/main/java/net/snowflake/ingest/utils/IcebergDataTypeParser.java
Comment (sfc-gh-hmadan) [inline src/main/java/net/snowflake/ingest/utils/IcebergDataTypeParser.java:77]: its unclear why you made this change. I see the only difference is in the avro-sanitization of the column name. Is that logic causing issues? 
Copy-pasting code from the iceberg-parquet library will need to consider licensing. IANAL Apache 2.0 requires adding the license as well as a summary of the modifications made while copying their code.

So if we really need to do this, I'd create an explicit TypeToMessageType file, add the proper license attribution at the top, and list what we've modified again in the file header.

My vote is to not do this at all.

If you want to keep the original column name around why not add a new field/property into ParquetTypeInfo, and pass the column name in there from ParquetTypeGenerator.generateColumnParquetTypeInfo ?

File: src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergNumericTypesIT.java
Comment (sfc-gh-hmadan) [inline src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergNumericTypesIT.java:37]: A fixed seed will imply the exact set of values being tested everytime. Why not have a randomly generated seed and plug that in here.
The reason to explicitly generate a seed and plug it in is - we can now log this seed / print it out in failure messages, and accurately reproduce any failures locally.

If you just do new Random(), reproducing a one-off error will be next to impossible.

File: src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergNumericTypesIT.java
Comment (sfc-gh-hmadan) [inline src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergNumericTypesIT.java:420]: this makes it impossible to reproduce failures locally. Why do we need this randomNumeric when we already generated a random number for intpart and floatpart using the seeded RNG in lines 412 and 413?

File: src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergNumericTypesIT.java
Comment (sfc-gh-alhuang) [reply src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergNumericTypesIT.java]: Switched to random seed.

File: src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergNumericTypesIT.java
Comment (sfc-gh-alhuang) [reply src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergNumericTypesIT.java]: The length of decimal is greater than int/long.max, thus we need string generator for this. Switch to generator with consistent seed.

File: src/main/java/net/snowflake/ingest/utils/IcebergDataTypeParser.java
Comment (sfc-gh-alhuang) [reply src/main/java/net/snowflake/ingest/utils/IcebergDataTypeParser.java]: Reverted.

File: src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergNumericTypesIT.java
Comment (sfc-gh-hmadan) [inline src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergNumericTypesIT.java:48]: please use seed instead of 123

File: src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergNumericTypesIT.java
Comment (sfc-gh-hmadan) [inline src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergNumericTypesIT.java:421]: not sure i fully understand what this is trying to do (comments will help?). Lets get this merged in to unblock KC folks and I'll follow up with you offline on this method.

**Review by sfc-gh-hmadan — APPROVED** [review]
LGTM, test code we'll discuss after this gets merged.

File: src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergNumericTypesIT.java
Comment (sfc-gh-alhuang) [reply src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergNumericTypesIT.java]: Use seed instead.

**Pull request merged**
